### PR TITLE
day06 homework

### PR DESCRIPTION
**Pull request by kghite**

### CODE REVIEW

File: day06/README.md [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
-In `MyTreeMap`:
+---
+---
 
-- Fill in the body of `findNode(Object target)`
+## Learning Goals
 
-- Fill in the body of `containsValue(Object target)`
+After completing this assignment you should be able to:
+* Understand the key concepts of binary search trees
```
Comment (davidabrahams) [inline day06/README.md]: What are these?

File: day06/README.md [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+## Assignment
 
-- Fill in the body of `keySet()`
+Finish the implementation of MyTreeMap (ch. 17.5):
+* Fill in the body of `findNode(Object target)`
+* Fill in the body of `containsValue(Object target)`
+* Fill in the body of `putHelper(Node node, K key, V value)`
+* Fill in the body of `keySet()`
 
+Draw a balanced binary search tree using the nodes [17, 19, 12, 76, 9, 50, 72, 23, 54, 14, 67].
```
Comment (thecardkid) [inline day06/README.md]: add statement saying that add this list in order. remove "balanced"

File: day06/README.md [hunk 3]
```diff
@@ -1,14 +1,32 @@
...
+---
+---
 
-- Fill in the body of `findNode(Object target)`
+## Learning Goals
 
-- Fill in the body of `containsValue(Object target)`
+After completing this assignment you should be able to:
+* Understand the key concepts of binary search trees
+* Analyze the efficiency differences of binary search trees
```
Comment (davidabrahams) [inline day06/README.md]: differences b/w what? BST and Hashmap?

File: day06/README.md [hunk 4]
```diff
@@ -1,14 +1,32 @@
...
+* Identify when utilizing a binary search tree is effective
```
Comment (davidabrahams) [inline day06/README.md]: a BST or a TreeMap? I feel like we should focus on TreeMaps, not a generalization of BSTs.

File: day06/README.md (hunk 2 above)
Comment (davidabrahams) [inline day06/README.md]: Given an array, output if the corresponding tree is a BST

File: day06/README.md [hunk 5]
```diff
@@ -1,14 +1,32 @@
...
+Draw a balanced binary search tree using the nodes [17, 19, 12, 76, 9, 50, 72, 23, 54, 14, 67].
+
+Briefly explain the advantages and disadvantages of using a BST rather than a hash table.
+
+## Problems
+
+Given a tree, write a function to determine if it is a binary search tree.
+
+Choose one of the following problems:
+* Card Shuffle - Fill in the function definitions in CardShuffle.java following the instructions in the file.
```
Comment (thecardkid) [inline day06/README.md]: Remove these two challenges. keep the "determine if BST", but add more specifications (e.g. your function will take in a 1d array, some elements can be null, will return true or false. Give examples, and you should write a test bench that tests 3 arrays)

File: day06/README.md [hunk 6]
```diff
@@ -1,14 +1,32 @@
...
+* Fill in the body of `findNode(Object target)`
+* Fill in the body of `containsValue(Object target)`
+* Fill in the body of `putHelper(Node node, K key, V value)`
+* Fill in the body of `keySet()`
 
+Draw a balanced binary search tree using the nodes [17, 19, 12, 76, 9, 50, 72, 23, 54, 14, 67].
+
+Briefly explain the advantages and disadvantages of using a BST rather than a hash table.
+
+## Problems
```
Comment (thecardkid) [inline day06/README.md]: Change "Problems" to "Extension"

File: day06/README.md (hunk 3 above)
Comment (thecardkid) [inline day06/README.md]: Change to "Analyze the efficiency of common binary search tree operations"

**Pull request merged**